### ISSUE TITLE
Pmag funtion to obtain rotation angles from rotation vectors

### ISSUE THREAD

**Issue by Caoimhinmg** [PmagPy/PmagPy#144]
So the idea is that there are dogeo and dotilt functions that take the original vector and rotation angles, turn those angles into a rotation matrix and then return the rotated vector. I'm wondering if there is a function in pmag that takes in the original vector and rotated vector and solves backwards for the rotation angles used to make the matrix? If not does someone know the appropriate equations to get azimuth and dip out of the origional vector and the rotated one? @ltauxe? I tried solving backwards myself but gave up after half an hour when I realized the pmag.dogeo function translated the rotation matrix into Cartesian and I got sick of trig. Though as far as I can tell there is a possible solution as there are 3 equations (3x3 matrix multiplied by a 3x1 vector so 3 equations) and 2 unknowns (azimuth and dip).

The application of this function is dominantly for the conversion scripts I've been working on. A number of data formats outside our own save dec, inc, geo_dec, geo_inc, tilt_dec, tilt_inc and it would be good not to loose the orientation data bundled in those different vectors when translating to MagIC for upload.

**Comment by ltauxe** [comment]
I did write one at some point but that is unfortunately a very unstable process.  I could hunt down what i did if you like. 
lisa

**Comment by Caoimhinmg** [comment]
It does seem rather tricky. I'd appreciate it greatly if you would point me in the right direction here because with that I theoretically could extract orientation data from most magnetometer files as they nearly all either store the orientation angles or the three dec inc pairs.

**Comment by Caoimhinmg** [comment]
Found it @ltauxe, Looks like it's called get_azpl and get_tilt. Completely missed it when I looked through the function names in pmag.py, but it is used in generic_magic.py (last file I worked on of course). Interestingly it is an iterative solution, guesses a azimuth and plunge then checks if it's right and goes through all possibilities. Out of curiosity has no one done an analytical solution to this problem? It would be a lot faster and more accurate that way. Well anyway I'm going to close the issue as this already exists.
